aten/src/ATen/native/cuda/Distributions.cu

```diff
@@ -724,13 +724,34 @@ Tensor& normal_out_cuda(Tensor& output, double mean, const Tensor& std, Generato
 }
 
 Tensor& normal_out_cuda(Tensor& output, const Tensor& mean, const Tensor& std, Generator* gen) {
+  bool expandable = are_expandable(mean.sizes(), std.sizes());
+  // allow resize if output is an empty tensor
+  if (output.sizes().equals({0})) {
+    if (expandable) {
+      auto shape = at::infer_size(mean.sizes(), std.sizes());
+      at::native::resize_(output, shape, LEGACY_CONTIGUOUS_MEMORY_FORMAT);
```

Comment:
you don't need to use the LEGACY_CONTIGUOUS_MEMORY_FORMAT assuming normal_cuda_, mul_, add_ support any memory format (you can just check or ask @VitalyFedyunin).

Reply:
Will check with Vitaly. Using LEGACY_CONTIGUOUS_MEMORY_FORMAT here is because the output tensor is empty at this point, we actually don't know what kind of memory format it should be allocated. So probably LEGACY_CONTIGUOUS_MEMORY_FORMAT is the best choice. And all the non-out version normal_cuda function all allocate output tensor using LEGACY_CONTIGUOUS_MEMORY_FORMAT.  I think this is consistent and backward compatible.

Reply:
but LEGACY_CONTIGUOUS_MEMORY_FORMAT is exactly equivalent to CONTIGUOUS_FORMAT, it just means we haven't actually checked that it's safe to make it not contiguous.